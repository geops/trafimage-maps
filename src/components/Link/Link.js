import React from 'react';
import PropTypes from 'prop-types';
import { ReactComponent as LinkIcon } from './Link.svg';

import './Link.scss';

const propTypes = {
  href: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

const defaultProps = {};

const Link = ({ href, children }) => (
  <a className="wkp-link" href={href} rel="noopener noreferrer" target="_blank">
    {children}
    <LinkIcon />
  </a>
);

Link.propTypes = propTypes;
Link.defaultProps = defaultProps;

export default Link;
