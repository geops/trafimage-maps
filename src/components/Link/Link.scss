.wkp-link {
  & > svg {
    padding-left: 5px;
  }
}
