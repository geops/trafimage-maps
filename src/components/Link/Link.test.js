import 'jest-canvas-mock';
import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Link from '.';

configure({ adapter: new Adapter() });

describe('Link', () => {
  test('should match snapshot.', () => {
    const component = renderer.create(
      <Link href="https://geops.de/">geOps</Link>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
