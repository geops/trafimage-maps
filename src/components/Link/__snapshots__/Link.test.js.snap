// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should match snapshot. 1`] = `
<a
  className="wkp-link"
  href="https://geops.de/"
  rel="noopener noreferrer"
  target="_blank"
>
  geOps
  <svg>
    Link.svg
  </svg>
</a>
`;
